x-pack/plugins/observability/public/hooks/slo/use_clone_slo.ts

```diff
@@ -5,83 +5,28 @@
  * 2.0.
  */
 
-import { IHttpFetchError, ResponseErrorBody } from '@kbn/core/public';
-import { i18n } from '@kbn/i18n';
-import type { CreateSLOInput, CreateSLOResponse, FindSLOResponse } from '@kbn/slo-schema';
-import { QueryKey, useMutation, useQueryClient } from '@tanstack/react-query';
-import { v4 as uuidv4 } from 'uuid';
+import { encode } from '@kbn/rison';
+import { useCallback } from 'react';
+import { SLOWithSummaryResponse } from '@kbn/slo-schema';
+import { paths } from '../../../common/locators/paths';
 import { useKibana } from '../../utils/kibana_react';
-import { sloKeys } from './query_key_factory';
-
-type ServerError = IHttpFetchError<ResponseErrorBody>;
 
 export function useCloneSlo() {
   const {
-    http,
-    notifications: { toasts },
+    http: { basePath },
+    application: { navigateToUrl },
   } = useKibana().services;
-  const queryClient = useQueryClient();
 
-  return useMutation<
-    CreateSLOResponse,
-    ServerError,
-    { slo: CreateSLOInput; originalSloId?: string },
-    { previousData?: FindSLOResponse; queryKey?: QueryKey }
-  >(
-    ['cloneSlo'],
-    ({ slo }: { slo: CreateSLOInput; originalSloId?: string }) => {
-      const body = JSON.stringify(slo);
-      return http.post<CreateSLOResponse>(`/api/observability/slos`, { body });
+  return useCallback(
+    (slo: SLOWithSummaryResponse) => {
+      navigateToUrl(
+        basePath.prepend(
+          paths.observability.sloCreateWithEncodedForm(
+            encode({ ...slo, name: `[Copy] ${slo.name}`, id: undefined })
```

Comment:
@shahzad31 Just out of curiosity in the previous implementation we were using `id: uuidv4()`. I guess we are calling `uuidv4()` when clicking on the `Create SLO` button, right?

Reply:
yeah because we were actually creating an SLO in memory. In new implementation we are just redirecting user to the pre filled form.